### ISSUE TITLE
tv.teads.sdk.adContent.views.componentView.EndScreenView.getVisibility()' on a null object reference

### ISSUE THREAD

**Issue by wliadmin** [teads/TeadsSDK-android#117]
I have got the below crash log on below version
- implementation ('tv.teads.sdk:androidsdk:2.4.14:fullRelease@aar') {
        transitive = true;
}

- I have reproduced this error on device Lenovo Vibe k5 plus (Android 5.1.1 API 22)

- Also I have added below configuration on gradle file
minSdkVersion = 16
compileSdkVersion = 28
targetSdk = 23

- Used classpath 'com.android.tools.build:gradle:3.1.3'

**Crash log:**
java.lang.NullPointerException: Attempt to invoke virtual method 'int tv.teads.sdk.adContent.views.componentView.EndScreenView.getVisibility()' on a null object reference
        at tv.teads.sdk.adContent.views.AdContentView.endScreenIsVisible(AdContentView.java:822)
        at tv.teads.sdk.adContent.video.VideoAdContent.c(VideoAdContent.java:525)
        at tv.teads.sdk.adContainer.AdContainer.a(AdContainer.java:172)
        at tv.teads.sdk.adContainer.SimpleAdContainer.k(SimpleAdContainer.java:176)
        at tv.teads.sdk.adContainer.SimpleAdContainer.a(SimpleAdContainer.java:143)
        at tv.teads.sdk.publisher.BaseTeadsAd.adDidLoad(BaseTeadsAd.java:544)
        at tv.teads.sdk.adContent.video.VideoAdContent.c(VideoAdContent.java:686)
        at tv.teads.sdk.adContent.video.ui.player.nativePlayer.exoplayer.TeadsExoPlayer.onPlayerStateChanged(TeadsExoPlayer.java:506)
        at tv.teads.android.exoplayer2.ExoPlayerImpl.handleEvent(ExoPlayerImpl.java:363)
        at tv.teads.android.exoplayer2.ExoPlayerImpl$1.handleMessage(ExoPlayerImpl.java:98)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:135)
        at android.app.ActivityThread.main(ActivityThread.java:5268)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:902)

**Comment by HugoGresse** [comment]
hi @wliadmin 
This was solved in `2.5.12`

    compile ('tv.teads.sdk:androidsdk:2.5.12:fullRelease@aar') {
        transitive = true;
    }

Let me know if you have another issue, in the meantime I'm closing this.

**Comment by wliadmin** [comment]
I have upgraded the same version but I think same crash happened on some other way 

And crash reproduced on Samsung Galaxy S4(Android 5.0 API 21)
**Crash:**
  ```
 java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.FrameLayout.setVisibility(int)' on a null object reference
        at tv.teads.sdk.adContent.views.AnimatedAdContentView.expand(AnimatedAdContentView.java:158)
        at ch.demo.ads.TeadsAdHandler.openInRead(TeadsAdHandler.java:206)
        at ch.demo.ads.TeadsAdHandler.access$1100(TeadsAdHandler.java:17)
        at ch.demo.ads.TeadsAdHandler$AdEventListener.teadsAdWillExpand(TeadsAdHandler.java:132)
        at tv.teads.sdk.publisher.BaseTeadsAd.adWillExpand(BaseTeadsAd.java:638)
        at tv.teads.sdk.adContent.AdContent.u(AdContent.java:248)
        at tv.teads.sdk.adContent.video.VideoAdContent.u(VideoAdContent.java:545)
        at tv.teads.sdk.adContainer.SimpleAdContainer.j(SimpleAdContainer.java:129)
        at tv.teads.sdk.publisher.BaseTeadsAd.adRequestShowContainer(BaseTeadsAd.java:667)
        at tv.teads.sdk.adContent.video.VideoAdContent.V(VideoAdContent.java:1066)
        at tv.teads.sdk.adContent.video.PlayerState.a(PlayerState.java:77)
        at tv.teads.sdk.adContent.video.VideoAdContent.W(VideoAdContent.java:1082)
        at tv.teads.sdk.adContent.video.VideoAdContent.c(VideoAdContent.java:695)
        at tv.teads.sdk.adContent.video.ui.player.nativePlayer.exoplayer.TeadsExoPlayer.onPlayerStateChanged(TeadsExoPlayer.java:506)
        at tv.teads.android.exoplayer2.ExoPlayerImpl.handleEvent(ExoPlayerImpl.java:363)
        at tv.teads.android.exoplayer2.ExoPlayerImpl$1.handleMessage(ExoPlayerImpl.java:98)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:145)
        at android.app.ActivityThread.main(ActivityThread.java:5951)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1388)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1183)
```
And i have used below code inside TeadsAdHandler>>openInRead() method.
 ```
private void openInRead() {
        if (mTeadsView == null) {
            mAdState.mAdViewHaveToBeOpen = true;
            return;
        }

        //Update the TeadsAdView to match the ViewGroup parent
        mTeadsView.updateSize(mListView);
        mTeadsView.setCollapsed();

        if (!mAdState.mAdViewHaveToBeOpen) {
            mAdState.mIsAnimating = true;
            mTeadsView.expand(new TeadsAnimationListener(true));
        } else {
            mAdState.mAdViewHaveToBeOpen = false;
            mTeadsView.setVisibility(View.VISIBLE);
            mAdState.mIsOpen = true;
            mTeadsAd.adViewDidExpand();
        }
    }
```
